### PR TITLE
Clean up viz code

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao mashimiao.fnst@cn.fujitsu.com

### CODE REVIEW

**Comment by crosbymichael** [comment]
Do we know what docker version the 1.6 API was in?  I think it would be ok to remove this endpoint if the docker version is below 1.0

**Comment by cpuguy83** [comment]
1.0 was API version 1.12

**Comment by hqhq** [comment]
So after this PR, we won't be able to use `docker images --tree`? Do we have some docs tell people how we do it in another way? Like using plugin as we discussed?

I know it is tagged as deprecated for a long time, but still a lot of people object this remove. So I think we should make it smooth.

**Comment by jfrazelle** [comment]
maybe we could split it out into its own tool or something... only problem is someone will have to maintain it

**Comment by jfrazelle** [comment]
who loves viz alot

**Comment by jfrazelle** [comment]
I am +1 to this change

**Comment by jfrazelle** [comment]
we have warned forever it will be depreciated, it is time

**Comment by Mashimiao** [comment]
ping @jfrazelle , @cpuguy83 , @crosbymichael 
rebased

**Comment by LK4D4** [comment]
Whoa, weird fail on windows. ping @cpuguy83 is that a race?

**Comment by LK4D4** [comment]
As I see it's pretty easy to write tool around `images/json`. Just copy client code there. I never used viz myself, so I don't want to write it :)

**Comment by cpuguy83** [comment]
Hmm, yeah, seems like there must be another container sitting around called "testing"...

**Comment by LK4D4** [comment]
@cpuguy83 #12511 should help.

**Comment by Mashimiao** [comment]
ping @jfrazelle , @cpuguy83 , @crosbymichael 
rebased.
janky failed. I think that's because of push race

**Comment by Mashimiao** [comment]
rebased again.

File: api/client/images.go [hunk 1]
```diff
@@ -116,158 +44,84 @@ func (cli *DockerCli) CmdImages(args ...string) error {
...
 
-				for _, repotag := range image.RepoTags {
-					if repotag == matchName {
-						startImage = &image
-					}
-				}
-			}
-		}
+	images := []types.Image{}
+	err = json.NewDecoder(rdr).Decode(&images)
```
Comment (runcom) [inline api/client/images.go]:
```
if err := json.NewDecoder(rdr).Decode(&images); err != nil {
    return err
} 
```

Comment (Mashimiao) [inline api/client/images.go]: Thanks. Fixed

**Comment by jfrazelle** [comment]
I think a lot are +1 on this, can we get another rebase so sorry

**Comment by Mashimiao** [comment]
@jfrazelle, rebased again.

**Comment by jfrazelle** [comment]
LGTM

**Comment by tiborvass** [comment]
LGTM

**Comment by LK4D4** [comment]
If I understand right, viz existed only on client-side.

**Comment by LK4D4** [comment]
LGTM

**Comment by tiborvass** [comment]
Actually this doesn't need docs updates since it wasn't even documented, being a very early feature that was removed before Docker 1.0

**Comment by tiborvass** [comment]
LGTM

**Comment by tiborvass** [comment]
Thanks for your contribution @Mashimiao !

**Comment by justone** [comment]
@jfrazelle, @hqhq, @LK4D4, I broke out the viz code into a separate tool called [dockviz](https://github.com/justone/dockviz) a little while ago and have added visualizing containers too, so there is an alternative for people who find it useful.

(disclosure: I was the original author of the tree and graphviz code)

**Comment by hqhq** [comment]
Thanks @justone , that's exactly what I need.

**Comment by LK4D4** [comment]
@moxiegirl can we add @justone tool to documentation somewhere and reference to `docker images`.
@justone Thank you! This is so cool!

**Comment by moxiegirl** [comment]
@LK4D4 first you'll have to explain to me what just happened. I think an option was added but I can't figure it out from the code. Can you give me a command line and I'll test it out.

**Comment by thaJeztah** [comment]
@moxiegirl some commands were deprecated and now *removed* from Docker. The tool @justone created offers the functionality that was removed from docker as an external tool

**Comment by moxiegirl** [comment]
@thaJeztah Thanks, I was like --- I don't see a new option but maybe I missed something.  I'll create an issue to document @justone

**Pull request merged**